web/src/app/users/linkToMobile/LinkToMobile.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useState, ReactNode } from 'react'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  makeStyles,
+  isWidthDown,
+  DialogContentText,
+} from '@material-ui/core'
+import PhonelinkIcon from '@material-ui/icons/Phonelink'
+import SwipeableViews from 'react-swipeable-views'
+import { virtualize, bindKeyboard } from 'react-swipeable-views-utils'
+import gql from 'graphql-tag'
+import { useQuery, useMutation } from 'react-apollo'
+import DialogTitleWrapper from '../../dialogs/components/DialogTitleWrapper'
+import useWidth from '../../util/useWidth'
+import { styles as globalStyles } from '../../styles/materialStyles'
+import ClaimCodeDisplay from './ClaimCodeDisplay'
+import VerifyCodeFields from './VerifyCodeFields'
+import Spinner from '../../loading/components/Spinner'
+import SuccessAnimation from '../../util/animations/SuccessAnimation'
+import ErrorAnimation from '../../util/animations/ErrorAnimation'
+import { useDispatch, useSelector } from 'react-redux'
+import { urlParamSelector } from '../../selectors'
+import { setURLParam } from '../../actions'
+import DialogContentError from '../../dialogs/components/DialogContentError'
+
+interface SlideParams {
+  index: number
+  key: number
+}
+
+const VirtualizeAnimatedViews = bindKeyboard(virtualize(SwipeableViews))
+
+const useStyles = makeStyles((theme) => {
+  const { cancelButton } = globalStyles(theme)
+
+  return {
+    cancelButton,
+    dialog: {
+      height: 345,
+      justifyContent: 'space-between',
+    },
+    successContainer: {
+      height: '100%',
+      width: '100%',
+      display: 'flex',
+      alignItems: 'center',
+      textAlign: 'center',
+    },
+  }
+})
+
+const mutation = gql`
+  mutation {
+    createAuthLink {
+      id
+      claimCode
+      verifyCode
+    }
+  }
+`
+
+export const query = gql`
+  query authLinkStatus($id: ID!) {
+    authLinkStatus(id: $id) {
+      id
+      expiresAt
+      claimed
+      verified
+      authed
+    }
+  }
+`
+
+export default function LinkToMobile(): JSX.Element {
+  const classes = useStyles()
+  const width = useWidth()
+  const fullscreen = isWidthDown('md', width)
+  const [showDialog, setShowDialog] = useState(false)
+
+  // getter for error messages
+  const urlParams = useSelector(urlParamSelector)
+  const errorMsg = urlParams('error', '')
+
+  // setter for error messages
+  const dispatch = useDispatch()
+  const setErrorMessage = (value: string): void => {
+    dispatch(setURLParam('error', value))
+  }
```

Comment:
Why are we sending the error to the address bar? are we reloading the page?

Reply:
I did it mainly because prop drilling through a bunch of steps kinda bothered me to set error based on the different query/mutation results, but I can certainly think of a better tree structure for the components.